nau_openedx_extensions/coursecertificate/engine.py

```
@@ -0,0 +1,257 @@
+        This is the main entry point for the business logic.
+        """
+        service_name = service_config.get("service_name")
+        self.log(f"\n=== Processing service: {service_name} ===")
+
+        # Get dry_run from options
+        dry_run = options.get("dry_run", False)
+
+        # Determine days to process
+        days = options.get("days")
+        if days is None:
+            days = service_config.get("days", 7)
+
+        # Determine page size
+        page_size = options.get("page_size")
+        if page_size is None:
+            page_size = service_config.get("page_size", 1000)
+
+        self.log(f"Processing certificates from the last {days} days")
+        self.log(f"Page size: {page_size}")
+
+        # Get certificates
+        certificates_queryset = self.get_certificates_queryset(days)
+
+        # Apply service-specific filters
+        certificates_queryset = self.apply_filters(certificates_queryset, service_config)
+
+        # Paginate and process
+        paginator = Paginator(certificates_queryset, page_size)
+        total_count = paginator.count
+        processed_count = 0
+
+        self.log(f"Total certificates to process: {total_count}")
+
+        for page_num in paginator.page_range:
+            page = paginator.page(page_num)
+            certificates = page.object_list
+            processed_count += len(certificates)
+
+            self.log(f"Processing {processed_count} of {total_count} certificates")
+
+            if certificates:
+                # Convert to service format
+                certificates_data = self.convert_certificates_to_service_format(certificates, service_config)
+
+                if certificates_data:
+                    # Send to service
+                    success = self.send_certificates_to_service(service_config, certificates_data, dry_run)
+                    if not success:
+                        self.log(f"Failed to send page {page_num} to {service_name}")
```

~Add missing break?~ 

Let it crash. Throw an exception.

---

and throw an exception